metadata-service/factories/src/main/java/com/linkedin/metadata/boot/OnBootApplicationListener.java

```diff
@@ -15,18 +24,56 @@
  */
 @Slf4j
 @Component
-public class OnBootApplicationListener implements ApplicationListener<ContextRefreshedEvent> {
+public class OnBootApplicationListener {
 
   private static final String ROOT_WEB_APPLICATION_CONTEXT_ID = String.format("%s:", WebApplicationContext.class.getName());
 
+  private final CloseableHttpClient httpClient = HttpClients.createDefault();
+
+  private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+
   @Autowired
   @Qualifier("bootstrapManager")
   private BootstrapManager _bootstrapManager;
 
-  @Override
+  @Autowired
+  @Qualifier("configurationProvider")
+  private ConfigurationProvider provider;
+
+
+  @EventListener(ContextRefreshedEvent.class)
   public void onApplicationEvent(@Nonnull ContextRefreshedEvent event) {
+    log.warn("OnBootApplicationListener context refreshed! {} event: {}",
+        ROOT_WEB_APPLICATION_CONTEXT_ID.equals(event.getApplicationContext().getId()), event);
     if (ROOT_WEB_APPLICATION_CONTEXT_ID.equals(event.getApplicationContext().getId())) {
-      _bootstrapManager.start();
+      executorService.submit(isSchemaRegistryAPIServeletReady());
     }
   }
+
+  public Runnable isSchemaRegistryAPIServeletReady() {
+    return () -> {
+        final HttpGet request = new HttpGet(provider.getKafka().getSchemaRegistry().getUrl());
+        int timeouts = 30;
+        boolean openAPIServeletReady = false;
+        while (!openAPIServeletReady && timeouts > 0) {
+          try {
+            log.info("Sleeping for 1 second");
+            Thread.sleep(1000);
+            StatusLine statusLine = httpClient.execute(request).getStatusLine();
+            if (statusLine.getStatusCode() == HttpStatus.SC_OK) {
+              log.info("Connected!");
+              openAPIServeletReady = true;
+            }
+          } catch (IOException | InterruptedException e) {
+            log.info("Failed to connect to open servlet: {}", e.getMessage());
+          }
+          timeouts--;
+        }
+        if (!openAPIServeletReady) {
+          log.error("Failed to bootstrap DataHub, OpenAPI servlet was not ready after 30 seconds");
```

Comment:
This should be an exit condition. We can't allow GMS to start without running the bootstrap manager.